.github/workflows/ci-cd.yml

```
@@ -0,0 +1,148 @@
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  lint-backend:
+    name: Code Quality Check - Backend
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
```

actionlint (1.7.7)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)